src/mtcp/mtcp_restart.c

```diff
@@ -552,6 +561,11 @@ static void restorememoryareas(RestoreInfo *rinfo_ptr)
 
   RestoreInfo restore_info;
   mtcp_memcpy(&restore_info, rinfo_ptr, sizeof (restore_info));
+  if (rinfo_ptr->saved_brk != NULL); {
```

Comment:
Basically, I wanted to make the minimal change possible.  I didn't want to do a new analysis of the code for catching errors in `restore_brk()`.